finetune/eval/if_eval/rule.py

```diff
@@ -54,11 +52,21 @@ def __init__(self, rule_id: RuleId):
         self.rule_id = rule_id
 
     @abstractmethod
-    def get_prompt(self) -> str:
-        """Returns the prompt for this rule."""
+    def get_prompt(self, index: int) -> str:
```

Comment:
Since any rule that cares about the context needs to check that the index is valid, could we use a default = -1 here to make the other classes cleaner in usage?